Sources/Internal/DoubleLayoutPassHelpers.swift

```diff
@@ -0,0 +1,136 @@
+/// the label to size correctly.
+///
+/// `CalendarView` can leverage `UILabel`'s double layout pass behavior by calling into this extension. Under the hood, a sizing
+/// `UILabel` is used to get a second layout pass, giving `CalendarView` a chance to size itself knowing its final width.
+
+extension CalendarView {
+
+  func installDoubleLayoutPassSizingLabel() {
+    doubleLayoutPassSizingLabel.removeFromSuperview()
+    addSubview(doubleLayoutPassSizingLabel)
+    subviews.first.map(doubleLayoutPassSizingLabel.sendSubviewToBack(_:))
+
+    doubleLayoutPassSizingLabel.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate([
+      doubleLayoutPassSizingLabel.leadingAnchor.constraint(
+        equalTo: layoutMarginsGuide.leadingAnchor),
+      doubleLayoutPassSizingLabel.trailingAnchor.constraint(
+        equalTo: layoutMarginsGuide.trailingAnchor),
+      doubleLayoutPassSizingLabel.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+      doubleLayoutPassSizingLabel.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor),
+    ])
+  }
+
+  public override func invalidateIntrinsicContentSize() {
+    doubleLayoutPassSizingLabel.invalidateIntrinsicContentSize()
+  }
+
+  public override func setContentHuggingPriority(
+    _ priority: UILayoutPriority,
+    for axis: NSLayoutConstraint.Axis)
+  {
+    doubleLayoutPassSizingLabel.setContentHuggingPriority(priority, for: axis)
+  }
+
+  public override func setContentCompressionResistancePriority(
+    _ priority: UILayoutPriority,
+    for axis: NSLayoutConstraint.Axis)
+  {
+    doubleLayoutPassSizingLabel.setContentCompressionResistancePriority(priority, for: axis)
+  }
+
+}
+
+// MARK: - WidthDependentIntrinsicContentHeightProviding
+
+protocol WidthDependentIntrinsicContentHeightProviding: CalendarView {
+
+  func intrinsicContentSize(forHorizontallyInsetWidth width: CGFloat) -> CGSize
+
+}
+
+// MARK: - DoubleLayoutPassSizingLabel
+
+final class DoubleLayoutPassSizingLabel: UILabel {
+
+  // MARK: Lifecycle
+
+  init(provider: WidthDependentIntrinsicContentHeightProviding) {
+    self.provider = provider
+
+    super.init(frame: .zero)
+
+    numberOfLines = 0
+    isUserInteractionEnabled = false
+    isAccessibilityElement = false
+  }
+
+  @available(*, unavailable)
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: Internal
+
+  override var intrinsicContentSize: CGSize {
+    guard let provider = provider else {
+      preconditionFailure(
+        "The sizing label's `provider` should not be `nil` for the duration of the its life")
+    }
+    if preferredMaxLayoutWidth == 0 {
+      return super.intrinsicContentSize
+    } else {
+      return provider.intrinsicContentSize(forHorizontallyInsetWidth: preferredMaxLayoutWidth)
+    }
+  }
+
+  override func draw(_: CGRect) {
```

Comment:
I probably mentioned this when implementing in the Airbnb app, but are we sure we need these empty overrides? There's the possibility that they impact performance https://stackoverflow.com/questions/18748276/why-an-empty-implementation-of-drawrect-will-adversely-affect-performance-durin

Perhaps we could just keep the label hidden.

Reply:
Ah yeah that's simpler. If people really go digging through the hierarchy to grab this label and set it visible, that's not really our problem 😛